src/main/scala/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkConnector.scala

```diff
@@ -1 +1,111 @@
-package com.microsoft.azure.cosmosdb.kafka.connect.sink
\ No newline at end of file
+package com.microsoft.azure.cosmosdb.kafka.connect.sink
+
+
+import java.util
+
+import com.microsoft.azure.cosmosdb.kafka.connect.sink.config.{CosmosDBConfig, CosmosDBSinkSettings}
+import com.microsoft.azure.cosmosdb.kafka.connect.sink.helpers._
+import com.microsoft.azure.cosmosdb._
+import com.microsoft.azure.cosmosdb.rx._
+
+import scala.util.{Failure, Success, Try}
+import scala.collection.JavaConverters._
+
+import com.typesafe.scalalogging.LazyLogging
+
+import org.apache.kafka.common.config.ConfigDef
+import org.apache.kafka.connect.connector.Task
+import org.apache.kafka.connect.sink.SinkConnector
+import org.apache.kafka.connect.errors.ConnectException
+
+
+
+class CosmosDBSinkConnector private[sink](builder: CosmosDBSinkSettings => AsyncDocumentClient) extends SinkConnector with LazyLogging {
+    private var configProps: util.Map[String, String] = _
+
+    override def version(): String = getClass.getPackage.getImplementationVersion
+
+    override def start(props: util.Map[String, String]): Unit = {
+        val config = Try(CosmosDBConfig(props)) match {
+            case Failure(f) => throw new ConnectException(s"Couldn't start Cosmos DB Sink due to configuration error: ${f.getMessage}", f)
+            case Success(c) => c
+        }
+
+        configProps = props
+
+        val settings = CosmosDBSinkSettings(config)
+        initCosmosDB(settings)
+
+        logger.info("Starting CosmosDBSinkConnector")
+    }
+
+    override def stop(): Unit = {
+        logger.info("Stopping CosmosDBSinkConnector")
+    }
+
+    override def taskClass(): Class[_ <: Task] = classOf[CosmosDBSinkTask]
+
+    override def taskConfigs(maxTasks: Int): util.List[util.Map[String, String]] = {
+        (1 to maxTasks).map(_ => this.configProps).toList.asJava
+    }
+
+    override def config(): ConfigDef = CosmosDBConfig.config
+
+    def initCosmosDB(settings: CosmosDBSinkSettings): Unit = {
+        implicit val documentClient: AsyncDocumentClient = AsyncDocumentClientProvider.get(settings)
+        val database = readOrCreateDatabase(settings)
```

Comment:
None of connectors create database/collection for a few reasons - there are many settings that would leak, this introduces an additional call, and there is a billing implication on RUs. Can we remove this?

Reply:
The other DocumentDB Connectors I used for inspiration did this. Also the MongoDB Connectors I looked at, did this. 

This is an option users can set. By Default it is set to FALSE.

What do you mean by "settings that would be leaked"?

Reply:
It will also only "introduce an additional call" once, when the Connector starts up, and not again. At least that was the intent.

Reply:
But I am impartial either way. It will certainly make for cleaner code if it was removed.